execution_engine/src/core/engine_state/step.rs

```diff
@@ -0,0 +1,97 @@
+use std::{collections::BTreeMap, fmt::Display, vec::Vec};
+
+use core::fmt;
+use uint::static_assertions::_core::fmt::Formatter;
+
+use casper_types::{bytesrepr, bytesrepr::ToBytes, Key, ProtocolVersion, PublicKey};
+
+use crate::shared::{newtypes::Blake2bHash, TypeMismatch};
+
+#[derive(Debug)]
+pub struct SlashItem {
+    pub validator_id: PublicKey,
+}
+
+impl SlashItem {
+    pub fn new(validator_id: PublicKey) -> Self {
+        Self { validator_id }
+    }
+}
+
+#[derive(Debug)]
+pub struct RewardItem {
+    pub validator_id: PublicKey,
+    pub value: u64,
+}
+
+impl RewardItem {
+    pub fn new(validator_id: PublicKey, value: u64) -> Self {
+        Self {
+            validator_id,
+            value,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct StepRequest {
+    pub parent_state_hash: Blake2bHash,
+    pub protocol_version: ProtocolVersion,
+
+    pub slash_items: Vec<SlashItem>,
+    pub reward_items: Vec<RewardItem>,
+    pub run_auction: bool,
```

Comment:
Will we ever run the step without running an auction?

Reply:
Discussed with Ed - the bool stays for now.

Reply:
Unclear.